src/components/layout/Navbar.js

```diff
@@ -232,18 +232,37 @@ const ServiceList = () => {
   const [services, isLoading, isError] = useFetchServiceExtensions()
   const [open, setOpen] = React.useState(true)
 
-  const handleClick = () => {
+  const handleToggleClick = (event) => {
+    event.stopPropagation()
     setOpen(!open)
   }
 
+  const handleToggleKeyDown = (event) => {
+    if (event.key === 'Enter' || event.key === ' ') {
+      event.stopPropagation()
+      setOpen(!open)
+    }
+  }
+
+  const handleNavigationClick = () => {
+    window.location.href = `#${SERVICES_PATH}`
```

Comment:
Why not inlined as href prop on Button?

Reply:
Using href directly on a button causes immediate navigation, which interrupts the folding action. The folding change is not visible because the page navigation occurs before the toggle state can update. Using a onClick function allows the toggle action to complete before navigating.

Reply:
Please rather use the [useNavigate](https://reactrouter.com/en/main/hooks/use-navigate) hook, i.e. `navigate(SERVICES_PATH)`.